### PR TITLE
Remove ProgressMeter

### PR DESCRIPTION
**Pull request by IAlibay**
Towards #2739 

Changes made in this Pull Request:
 - Removes deprecated ProgressMeter

Questions:
 - Do we need a versionchanged in lib.log?

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/lib/log.py [hunk 1]
```diff
@@ -80,12 +80,11 @@
 .. autogenerated, see Online Docs
 
 """
-from __future__ import print_function, division, absolute_import
+from __future__ import absolute_import
```
Comment (orbeckst) [inline package/MDAnalysis/lib/log.py:84]: Remove?

**Comment by orbeckst** [comment]
And yay, -282 lines of code!

Btw, if there’s widespread support for not adding versionchanged for dropped content then say so. I am just thinking that the technical docs can be technical and have useful information. Anything we put in the docs now will hopefully allow users to come to their own conclusions as opposed to us explaining.

**Comment by IAlibay** [comment]
> And yay, -282 lines of code!
> 
> Btw, if there’s widespread support for not adding versionchanged for dropped content then say so. I am just thinking that the technical docs can be technical and have useful information. Anything we put in the docs now will hopefully allow users to come to their own conclusions as opposed to us explaining.

I think my main thing was that I wasn't too sure where the versionchanged should go, I've just placed one under "13.2.6.2. Other functions and classes for logging purposes" hopefully it looks reasonable.

File: package/MDAnalysis/lib/log.py (hunk 1 above)
Comment (IAlibay) [reply package/MDAnalysis/lib/log.py:88]: Sure, I'll remove all the py2 stuff from lib at the same time.

**Pull request merged**